edx_when/models.py

```diff
@@ -75,6 +90,22 @@ def __str__(self):
         # TODO: return a string appropriate for the data fields
         return '<ContentDate, ID: {}>'.format(self.id)
 
+    def schedule_for_user(self, user):
+        """
+        Return the schedule for the supplied user that applies to this piece of content.
+        """
+        if isinstance(user, int):
+            if Schedule is None:
+                return None
+
+            return Schedule.objects.find_one(enrollment__user__id=user, enrollment__course__id=self.course_id)
```

Comment:
Why are these two branches written differently? This one could be something like `Enrollments.objects.find_one(user=, course=).schedule` or the second could be done with the same `Schedule is None` check and a Schedule query.

Like, it doesn't super matter, it's just odd to me that one path would early exit for Schedule is None, but the other doesn't, preferring to try/except.

Reply:
Yeah, I see your point. I think I wrote them differently mainly because I prefer explicit `is None` checks when I'm setting something explicitly to `None`, but don't really like `hasattr` checks (which is what you'd need below for the `enrollments` branch). But now that you mention it, the `except AttributeError` could potentially be catching multiple different missing attributes (and potential future misspellings), so I might tweak this.